src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/OrganizationUserUserDetailsQuery.cs

```diff
@@ -37,4 +50,45 @@ public async Task<IEnumerable<OrganizationUserUserDetails>> GetOrganizationUserU
                 return o;
             });
     }
+
+    /// <summary>
+    /// Get the organization user user details, two factor enabled status, and
+    /// claimed status for the provided request.
+    /// </summary>
+    /// <param name="request">Request details for the query</param>
+    /// <returns>List of OrganizationUserUserDetails</returns>
+    public async Task<IEnumerable<(OrganizationUserUserDetails, bool, bool)>> Get(OrganizationUserUserDetailsQueryRequest request)
+    {
+        var organizationUsers = await GetOrganizationUserUserDetails(request);
+
+        var organizationUsersTwoFactorEnabled = await _twoFactorIsEnabledQuery.TwoFactorIsEnabledAsync(organizationUsers);
```

Comment:
Please transform this result into a HashSet or similar so that we don't have nested loops below. (i.e. looping over this result inside the loop for each user.)